### PR TITLE
[OPEN_CONF]Add yang2spec lookup plugin

### PR DESCRIPTION
**Pull request by ganeshrn**
Depends-On: ansible-network/ansible-zuul-jobs#21

Fixes https://github.com/ansible-network/network-engine/issues/48

*  Generate a rules spec and json configuration schema
   from the input yang document.
*  The rules spec will be used to validate the json
   configuration to check if it adheres to the respective yang model.
*  The json configuration schema can be used as a reference to build
   json configuration which conforms with yang model.

### CODE REVIEW

**Review by trishnaguha — CHANGES_REQUESTED** [review]
Please add integration test for the plugin in `tests/` and add the plugin on README https://github.com/ansible-network/network-engine#lookup.

File: lookup_plugins/yang2spec.py [hunk 1]
```diff
@@ -0,0 +1,1556 @@
+#
```
Comment (gundalow) [inline lookup_plugins/yang2spec.py]: Please add changelog fragment

File: lookup_plugins/yang2spec.py [hunk 2]
```diff
@@ -0,0 +1,1556 @@
...
+#  Copyright 2018 Red Hat | Ansible
+#
```
Comment (gundalow) [inline lookup_plugins/yang2spec.py]: Need some tests adding for this new lookup plugin.

**Comment by gundalow** [comment]
Various sanity checks have failed https://ansible.softwarefactory-project.io/logs/85/85/1fccc223ee6ecd8e1001ba66b436953d8a6b560e/check/ansible-test-sanity/b6ec3ed/job-output.txt.gz#_2018-05-24_08_19_39_057888

 Including not using our version of `six`

File: lookup_plugins/yang2spec.py [hunk 3]
```diff
@@ -0,0 +1,1556 @@
...
+        with respective yang model. It also outputs the  configuration schema in json format and can
+        be used as reference to build input json configuration.
+    options:
+      _terms:
+        description: The path points to the location of the top level yang module which
+        is to be transformed into to Ansible spec.
+        required: True
+      search_path:
+        description:
+          - The path is a colon (:) separated list of directories to search for imported yang modules
```
Comment (gundalow) [inline lookup_plugins/yang2spec.py]: `is a colon C(:) separated`

File: lookup_plugins/yang2spec.py [hunk 4]
```diff
@@ -0,0 +1,1556 @@
...
+            the current directory.
+        required: false
+"""
+
+EXAMPLES = """
+- name: Get interface yang spec
+  set_fact:
+    interfaces_spec: "{{ lookup('yang2spec', 'openconfig/public/release/models/interfaces/openconfig-interfaces.yang',
+                            search_path='openconfig/public/release/models:pyang/modules/') }}"
+"""
```
Comment (gundalow) [inline lookup_plugins/yang2spec.py:38]: Would it be worth giving an example of how to use `interfaces_spec`?

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lookup_plugins/yang2spec.py (hunk 1 above)
Comment (ganeshrn) [reply lookup_plugins/yang2spec.py]: Will add the changelog fragment after this PR is merged as the changelog file format name starts with the version number in which the PR is added eg: `v252-filter-plugins.yaml`

File: lookup_plugins/yang2spec.py (hunk 4 above)
Comment (ganeshrn) [reply lookup_plugins/yang2spec.py:38]: The example above indicates how to use yang2spec plugin. The example of how to use the output of yang2spec will be added in openconfig role

**Comment by ganeshrn** [comment]
@trishnaguha @gundalow Updated testcases

**Comment by trishnaguha** [comment]
recheck

**Review by trishnaguha — CHANGES_REQUESTED** [review]
please import the test in `tests/test.yml`

**Comment by ganeshrn** [comment]
@trishnaguha Fixed your review comments

**Comment by trishnaguha** [comment]
@ganeshrn Thanks. Please resolve the conflict and we need to wait for https://github.com/ansible-network/ansible-zuul-jobs/pull/21 to be merged for CI to pass.

**Comment by trishnaguha** [comment]
recheck

**Comment by trishnaguha** [comment]
Python3 failure:
```
TASK [yang2spec : Convert interface yang module to spec] ***********************
2018-06-06 09:15:01.326338 | container | task path: /home/zuul-worker/src/github.com/ansible-network/network-engine/tests/yang2spec/yang2spec/tasks/basic.yaml:2
2018-06-06 09:15:01.918652 | container | fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'yang2spec'. Error was a <class 'AttributeError'>, original message: 'str' object has no attribute 'decode'"}
```

**Review by trishnaguha — APPROVED** [review]

**Pull request merged**